src/commands/force/mdapi/deploy/report.ts

```diff
@@ -82,20 +82,23 @@ export class Report extends DeployCommand {
     );
 
     if (this.isAsync) {
-      this.deployResult = await this.report(this.getFlag<string>('jobid'));
+      this.deployResult = await this.report(deployId);
       return;
     }
 
-    this.displayDeployId(deployId);
-
     const deploy = this.createDeploy(deployId);
     if (!this.isJsonOutput()) {
       const progressFormatter: ProgressFormatter = env.getBoolean('SFDX_USE_PROGRESS_BAR', true)
         ? new DeployProgressBarFormatter(this.logger, this.ux)
         : new DeployProgressStatusFormatter(this.logger, this.ux);
       progressFormatter.progress(deploy);
     }
-    this.deployResult = await deploy.pollStatus({ frequency: Duration.milliseconds(500), timeout: waitDuration });
+
+    try {
+      await deploy.pollStatus({ frequency: Duration.milliseconds(500), timeout: waitDuration });
+    } finally {
+      this.deployResult = await this.report(deployId);
```

Comment:
Couldn't we run into a scenario where `pollStatus()` is successful but then we run `this.report()` and get a connection error?

Reply:
Good point.  `source:deploy` already was already open to that potential issue.

Would you rather try to set `this.deployResult` from pollStatus, and then finally would call this.report only if needed?

The problem is this.report has all kinds of side effects beyond checking status (handling undefined id from the stash, printing to the terminal, passing the componentSet part to SDR/DeployResult's constructor, etc).